### ISSUE TITLE
NullReferenceException when using spread operators

### ISSUE THREAD

**Issue by DavidKOneAdvanced** [trullock/NUglify#326]
Using Nuglify 1.20.0 with the following JS:

```
var array = [{ name: "hello" }, { name: "hello2" }]
let sObj = {}
sObj = {
    ...array.find(s2 => s2.name === "hello")
}
```

Nuglify crashes with a NullReferenceException:

```
   at NUglify.JavaScript.Syntax.FunctionScope.DefineFunctionExpressionName()
   at NUglify.JavaScript.Syntax.FunctionScope.DeclareScope()
   at NUglify.JavaScript.Visitors.ResolutionVisitor.CreateFields(ActivationObject scope)
   at NUglify.JavaScript.Visitors.ResolutionVisitor.CreateFields(ActivationObject scope)
   at NUglify.JavaScript.Visitors.ResolutionVisitor.Apply(BlockStatement block, ActivationObject scope, JSParser parser)
   at NUglify.JavaScript.JSParser.InternalParse()
   at NUglify.JavaScript.JSParser.Parse(DocumentContext sourceContext)
   at NUglify.Uglify.Js(String source, String fileName, CodeSettings codeSettings)
```

As far as I can see that's valid JS and other minifiers seem to work.

**Comment by trullock** [comment]
Can you try replacing the lambda with a function, just to see if that's the cause?

**Comment by DavidKOneAdvanced** [comment]
@trullock I've tried:

```
var array = [{ name: "hello" }, { name: "hello2" }]
let sObj = {}
sObj = {
    ...array.find(function(s2) { return s2.name === "hello" })
}
```

Still failing. I think it's the spread operator coming from a .find that's tripping it up.

**Comment by trullock** [comment]
Released PR in 1.21.4

**Comment by trullock** [comment]
Thanks for the PR, feel free to look at any of the others ;)
